### PR TITLE
Add primary & secondary connection string to 'azurerm_iothub_dps_shared_access_policy'

### PR DESCRIPTION
**Pull request by brunhil**
Add **primary_connection_string** and **secondary_connection_string** to _azurerm_iothub_dps_shared_access_policy_

--- PASS: TestAccAzureRMIotHubDpsSharedAccessPolicy_writeWithoutRead (151.97s)
--- PASS: TestAccAzureRMIotHubDpsSharedAccessPolicy_enrollmentReadWithoutRegistration (166.87s)
--- PASS: TestAccAzureRMIotHubDpsSharedAccessPolicy_enrollmentWriteWithoutOthers (205.74s)
--- PASS: TestAccAzureRMIotHubDpsSharedAccessPolicy_basic (273.52s)

### CODE REVIEW

File: azurerm/resource_arm_iothub_dps_shared_access_policy.go [hunk 1]
```diff
@@ -217,6 +234,14 @@ func resourceArmIotHubDPSSharedAccessPolicyRead(d *schema.ResourceData, meta int
 	d.Set("primary_key", accessPolicy.PrimaryKey)
 	d.Set("secondary_key", accessPolicy.SecondaryKey)
 
+	if err := d.Set("primary_connection_string", getSAPConnectionString(*iothubDps.Properties.ServiceOperationsHostName, keyName, *accessPolicy.PrimaryKey)); err != nil {
```
Comment (katbyte) [inline azurerm/resource_arm_iothub_dps_shared_access_policy.go]: We'll need to nil check these properties here:
```suggestion
	if props := iothubDps.Properties; props != nil {
		if hosts := props.ServiceOperationsHostName; hosts != nil
			if pkey = accessPolicy.PrimaryKey; pkey != nil {
				if err := d.Set("primary_connection_string", getSAPConnectionString(*iothubDps.Properties.ServiceOperationsHostName, keyName, *accessPolicy.PrimaryKey)); err != nil {
```

**Comment by brunhil** [comment]
Would it be easier if I created a new PR with the provider reorganization changes or pulled the changes in after the PR is approved @katbyte ?

File: azurerm/resource_arm_iothub_dps_shared_access_policy.go [hunk 2]
```diff
@@ -217,6 +234,26 @@ func resourceArmIotHubDPSSharedAccessPolicyRead(d *schema.ResourceData, meta int
...
+
+	if props := iothubDps.Properties; props != nil {
+		if hosts := props.ServiceOperationsHostName; hosts != nil {
+			if sKey := accessPolicy.SecondaryKey; sKey != nil {
+				if err := d.Set("secondary_connection_string", getSAPConnectionString(*iothubDps.Properties.ServiceOperationsHostName, keyName, *accessPolicy.SecondaryKey)); err != nil {
+					return fmt.Errorf("error setting `secondary_connection_string`: %v", err)
+				}
+			}
+		}
+	}
```
Comment (katbyte) [inline azurerm/resource_arm_iothub_dps_shared_access_policy.go]: could we make this:
```
	if props := iothubDps.Properties; props != nil {
		if host := props.ServiceOperationsHostName; hosts != nil {
			if pKey := accessPolicy.PrimaryKey; pKey != nil {
				if err := d.Set("primary_connection_string", getSAPConnectionString(*host, keyName, *pKey)); err != nil {
					return fmt.Errorf("error setting `primary_connection_string`: %v", err)
				}
			}
		
			if sKey := accessPolicy.SecondaryKey; sKey != nil {
				if err := d.Set("secondary_connection_string", getSAPConnectionString(*host, keyName, *sKey)); err != nil {
					return fmt.Errorf("error setting `secondary_connection_string`: %v", err)
				}
			}
		}
	}
```

**Comment by brunhil** [comment]
> @brunhil, Either or would work. Just one comment i've left inline that once addressed this'll be good to merge

End result for PR is ugly, but rebase is done :). Let me know your thoughts!

**Pull request merged**